File: 서인/CH06/2_괄호_문자_제거.js
function solution(s){  
    let answer = "";
    let stack=[];
    
    for (let x of s){
      if (x === "(") stack.push(x)
      if (stack.length === 0) answer += x
      if (x === ")") stack.pop(x)
    }

    return answer;
}

File: 종복/Java/Stack&Queue/2. 괄호 문자 제거.java
import java.util.*;
class Main {	
	public String solution(String str){
		String answer="";
        Stack<Character> stk = new Stack<>(); 
         
        for(char x: str.toCharArray()){
            if(x == '('){
                stk.push('(');
            }else if(x == ')'){
                stk.pop();
            }else{
                if(stk.isEmpty()){
                    answer += Character.toString(x);
                }
            }
        } 
     
		return answer;
	}

	public static void main(String[] args){
		Main T = new Main();
		Scanner kb = new Scanner(System.in);
		String str=kb.next();
		System.out.println(T.solution(str));
	}
}
